tests/test.cpp

```diff
@@ -4324,6 +4324,78 @@ void FlatbuffersIteratorsTest() {
 void FlatbuffersIteratorsTest() {}
 #endif
 
+void PrivateAnnotationsLeaks() {
+  // Simple schemas and a "has optional scalar" sentinal.
+  std::vector<std::string> schemas;
+  schemas.push_back(
+      "table Monster (private) { mana : int; } struct ABC (private) { mana: "
+      "int; }");
+  schemas.push_back("table Monster { mana : int; } struct ABC { mana: int; }");
+
+  schemas.push_back(
+      "table Monster (private) { mana : int; } struct ABC (private) { mana: "
+      "int; } union Any (private) { Monster, ABC }");
+
+  schemas.push_back(
+      "table Monster { mana : int; } struct ABC { mana: int; } union Any { "
+      "Monster, ABC }");
+  schemas.push_back(
+      "table Monster (private) { mana : int; } struct ABC (private) { mana: "
+      "int; } enum Race:byte (private) { None = -1, Human = 0, }");
+  schemas.push_back(
+      "table Monster { mana : int; } struct ABC { mana: int; } enum Race:byte "
+      "{ None = -1, Human = 0, }");
+  schemas.push_back(
+      "table Monster (private) { mana : int; } struct ABC (private) { mana: "
+      "int; } enum Race:byte (private) { None = -1, Human = 0, } union Any "
+      "(private) { Monster, ABC }");
+  schemas.push_back(
+      "table Monster { mana : int; } struct ABC { mana: int; } enum Race:byte "
+      "{ None = -1, Human = 0, } union Any { Monster, ABC }");
+
+  std::vector<std::string> failure_schemas;
+  failure_schemas.push_back(
+      "table Monster (private) { mana : int; } struct ABC { mana: int; }");
+  failure_schemas.push_back(
+      "table Monster (private) { mana : int; } struct ABC { mana: int; } union "
+      "Any { Monster, ABC }");
+  failure_schemas.push_back(
```

Comment:
why does this schema fail?

Monster is private but it's not being leaked - I think its leaked if its in a public field of another table or something

Reply:
Shouldn't it fail since `"Any { Monster, ABC }` leaks it, right?

Reply:
oh you're right, I read that wrong